archunit/src/main/java/com/tngtech/archunit/library/modules/ArchModules.java

```diff
@@ -0,0 +1,556 @@
+        }
+
+        /**
+         * Allows to customize each {@link ArchModule} {@link ArchModule#getName() name} by specifying a string pattern
+         * that defines how to derive the name from the {@link ArchModule.Identifier}.<br>
+         * In particular, the passed {@code namingPattern} may contain numbered placeholders like <code>${1}</code>
+         * to refer to parts from the {@link ArchModule.Identifier}. It may also contain the special placeholder
+         * <code>$@</code> to refer to the colon-joined form of the identifier.<br>
+         * Suppose the {@link ArchModule.Identifier} is {@code ["customer", "creation"]}, then the name could be derived as
+         * <pre><code>
+         * "Module[${1}/${2}]" -> would yield the name "Module[customer/creation]"
+         * "Module[$@]         -> would yield the name "Module[customer:creation]"
+         * </code></pre>
+         * Note that the derived name must be unique between all {@link ArchModule modules}.
+         *
+         * @param namingPattern A string naming pattern deriving the {@link ArchModule} {@link ArchModule#getName() name} from
+         *                      the {@link ArchModule.Identifier}
+         * @return A fluent API to further customize how to create {@link ArchModules}
+         */
+        @PublicAPI(usage = ACCESS)
+        public Creator deriveNameFromPattern(final String namingPattern) {
+            return new Creator(identifierAssociation, identifier -> {
+                String result = namingPattern;
+                for (int i = 1; i <= identifier.getNumberOfParts(); i++) {
+                    result = result
+                            .replace("$" + i, identifier.getPart(i))
+                            .replace("${" + i + "}", identifier.getPart(i))
+                            .replace("$@", joinIdentifier(identifier));
+                }
+                return result;
+            });
+        }
+
+        /**
+         * Allows to fully customize the {@link ArchModule.Descriptor} of the created {@link ArchModule}s. This allows to
+         * pass on meta-data from the contained classes additionally to any derived {@link ArchModule#getName() module name}.
+         * @param descriptorCreator A generic function specifying how to create the {@link ArchModule.Descriptor}
+         * @return A fluent API to further customize how to create {@link ArchModules}
+         * @param <D> The specific subtype of the {@link ArchModule.Descriptor} to create
+         */
+        @PublicAPI(usage = ACCESS)
+        public <D extends ArchModule.Descriptor> WithGenericDescriptor<D> describeBy(DescriptorCreator<D> descriptorCreator) {
+            return new WithGenericDescriptor<>(identifierAssociation, descriptorCreator);
+        }
+
+        /**
+         * @see WithGenericDescriptor#modularize(JavaClasses)
+         */
+        @PublicAPI(usage = ACCESS)
+        public ArchModules<?> modularize(JavaClasses classes) {
```

Comment:
As discussed, "modularize" for the action of creating modules from a set of elements seems "good enough" for now and the imperative form `partition(classes)` would otherwise also look somewhat strange in the base API :wink: